libsolidity/interface/StandardCompiler.cpp

```diff
@@ -162,11 +162,55 @@ Json::Value StandardCompiler::compileInternal(Json::Value const& _input)
 	if (!sources)
 		return formatFatalError("JSONError", "No input sources specified.");
 
+	Json::Value errors = Json::arrayValue;
+
 	for (auto const& sourceName: sources.getMemberNames())
 		if (sources[sourceName]["content"].isString())
 			m_compilerStack.addSource(sourceName, sources[sourceName]["content"].asString());
 		else if (sources[sourceName]["urls"].isArray())
-			return formatFatalError("UnimplementedFeatureError", "Input URLs not supported yet.");
+		{
+			unsigned count = sources[sourceName]["urls"].size();
+			for (auto const& url: sources[sourceName]["urls"])
+			{
+				count--;
+				ReadFile::Result result{false, "No read file callback supplied"};
+				try
+				{
+					cout << "reading " << url.asString() << endl;
+					if (m_readFile)
+						result = m_readFile(url.asString());
+				}
+				catch (Exception const& _exception)
+				{
+					errors.append(formatError(
```

Comment:
Do we have the same code for the import-triggered read callback? If yes, we should unify it, if no, we should also unify it :-)